### PR TITLE
Customizing Generators - issues #28

### PR DESCRIPTION
**Pull request by laeciofreitas**
vraptor-scaffold search for templates in other paths.

name_app/src/templates/controller/ controller.erb or controller_test.erb

name_app/src/templates/model/ ..

name_app/src/templates/repository/ ...

name_app/src/templates/engine/ new.erb ....

### CODE REVIEW

**Comment by rodolfoliviero** [comment]
this patch looks good. I will review asap. Thanks

**Comment by rodolfoliviero** [comment]
@laeciofreitas may you change the latest commits to another branch? 
If i merge it now i will include some features that are in progress.

File: lib/vraptor-scaffold/generators/scaffold/template_engine_generator/freemarker_generator/freemarker_generator.rb [hunk 1]
```diff
@@ -15,4 +11,13 @@ class FreemarkerGenerator < TemplateEngineGenerator
   def path
     "${base}/#{base_path}"
   end
+  
+  def template_path
+    "src/templates/engine"
```
Comment (rodolfoliviero) [inline lib/vraptor-scaffold/generators/scaffold/template_engine_generator/freemarker_generator/freemarker_generator.rb:16]: what about change this path to src/templates/views?

Comment (laeciofreitas) [inline lib/vraptor-scaffold/generators/scaffold/template_engine_generator/freemarker_generator/freemarker_generator.rb:16]: I think is better src/templates/views.

**Comment by laeciofreitas** [comment]
which branch to send the commits?

**Comment by rodolfoliviero** [comment]
Every time you open a pull request all the new commits that you push to branch automatically will be send to be merge. 
So i suggest you to create a new branch for every feature that you want to be merged.

In this case can you have a branch called 'customizing-generators'  only wich this commit https://github.com/laeciofreitas/vraptor-scaffold/commit/3a459122716cd8f452719155209e85ad1dcbcae2 

and commit to this branch only the changes about 'customizing generators' like change the engine source path to 'src/templates/views'.

So you can have another branch called ''new-attributes' or something like that and put the another commits there and when you finish this feature you open a new pull request form this branch

Now i will merge the first commit manually and ask you to open another pull request with the changes that i'd like?

Ok for you?

**Comment by rodolfoliviero** [comment]
I just pushed to master this commit 'support for customizing generators'  manually.
So closing this pull request 
Thanks.

**Comment by laeciofreitas** [comment]
also close the issue 28.

**Pull request closed without merging**